packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -3110,7 +3110,8 @@ function beginWork(
                 // been unsuspended it has committed as a resolved Suspense component.
                 // If it needs to be retried, it should have work scheduled on it.
                 workInProgress.effectTag |= DidCapture;
-                break;
+
+                return null;
```

Comment:
This revealed that if you do this, then it'll schedule work on the "children" of the Suspense boundary. Before it mounts. We then try to render into the children. The Suspense boundary hasn't changed, so we get here to bail out. The normal bail out tries to render children. The child is a DehydratedFragment that isn't a valid tag to render. (It has no beginWork). That causes an invariant.

We never expect to render these children so I just bail out by returning null instead.

Reply:
Some form of that would have made a nice inline comment explaining the significance of the return 😅